handlers.go

```diff
@@ -3153,6 +3156,75 @@ func (s *Server) handleAddContentsToCollection(c echo.Context, u *User) error {
 	return c.JSON(200, map[string]string{})
 }
 
+// handleCommitCollection godoc
+// @Summary      Produce a CID of the collection contents
+// @Description  This endpoint is used to save the contents in a collection, producing a top-level CID that references all the current CIDs in the collection.
+// @Param        coluuid     path     string  true     "coluuid"
+// @Tags         collections
+// @Produce      json
+// @Success      200  {object}  string
+// @Router       /collections/{collection-id}/commit [post]
+func (s *Server) handleCommitCollection(c echo.Context, u *User) error {
+	colid := c.Param("coluuid")
+
+	var col Collection
+	if err := s.DB.First(&col, "uuid = ? and user_id = ?", colid, u.ID).Error; err != nil {
+		return err
+	}
+
+	contents := []Content{}
+	if err := s.DB.Debug().
+		Model(CollectionRef{}).
+		Where("collection = ?", col.ID).
+		Joins("left join contents on contents.id = collection_refs.content").
+		Select("contents.*").
+		Scan(&contents).Error; err != nil {
+		return err
+	}
+
+	collectionNode := unixfs.EmptyDirNode()
+	for _, c := range contents {
+		collectionNode.AddRawLink(c.Name, &ipld.Link{
```

Comment:
`c.Name` is going to be different than the name of the thing within the collection, we need to make sure we use whatever is on the `CollectionRef` object in question here.

Reply:
This is just a name for the link. Also, `CollectionRef` doesn't really have a name for the content, it just has a path for it inside the collection and the ID pointing to the content in the table.